lib/cfg.h

```diff
@@ -181,6 +181,16 @@ cfg_is_config_version_older(GlobalConfig *cfg, gint req)
   return TRUE;
 }
 
+/* VERSION_LAST_SEMANTIC_CHANGE needs to be bumped in versioning.h whenever
+ * we add a conditional on the config version anywhere in the codebase.  The
+ * G_STATIC_ASSERT checks that we indeed did that */
+
+#define cfg_is_config_version_older(__cfg, __req) \
+  ({ \
+    /* check that VERSION_LAST_SEMANTIC_CHANGE is set correctly */ G_STATIC_ASSERT((__req) <= VERSION_LAST_SEMANTIC_CHANGE); \
```

Comment:
I didn't understood why is this a one liner, but then saw the make output.
Nice way of adding error message support. :)
```
../lib/cfg.h:190:68: note: in expansion of macro ‘G_STATIC_ASSERT’
     /* check that VERSION_LAST_SEMANTIC_CHANGE is set correctly */ G_STATIC_ASSERT((__req) <= VERSION_LAST_SEMANTIC_CHANGE); \
```